The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9886043878/job/27305102958) [(plain)](https://github.com/rust-lang/rust/commit/848c59165e0d8f3cec875cd662d8b154bf4d2914/checks/27305102958/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:692973e3d937129bcbf40652eb9f2f61becf3332)
Download action repository 'actions/upload-artifact@v4' (SHA:0b2256b8c012f0828dc542b3febcab082c67f72b)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.409   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.422 Collecting boolean-py==4.0
#13 0.425   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.435 Collecting chardet==5.1.0
---
#13 3.474 Building wheels for collected packages: reuse
#13 3.475   Building wheel for reuse (pyproject.toml): started
#13 3.807   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.808   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.809   Stored in directory: /tmp/pip-ephem-wheel-cache-lrs5qze5/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.811 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.834   Attempting uninstall: setuptools
#13 3.835     Found existing installation: setuptools 59.6.0
#13 3.836     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.087   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 5.154      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 88.6 MB/s eta 0:00:00
#13 5.209 Collecting filelock<4,>=3.12.2
#13 5.212   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 5.229 Collecting distlib<1,>=0.3.7
#13 5.232   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.238      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 118.1 MB/s eta 0:00:00
#13 5.266 Collecting platformdirs<5,>=3.9.1
#13 5.270   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.359 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.533 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 5.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      219072 kB
DirectMap2M:     6072320 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/8c39ac9eccdfaeff11bd3b6107817a81fe0a3a43/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-8c39ac9eccdfaeff11bd3b6107817a81fe0a3a43-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
fmt check
error: expected identifier, found keyword `as`
##[error]  --> /checkout/library/test/src/test_result.rs:96:28
   |
96 |         Some(0xC0000409u32 as i32) => TestResult::TrFailed,


error: expected one of `)`, `,`, `...`, `..=`, `..`, or `|`, found keyword `as`
   |
   |
96 |         Some(0xC0000409u32 as i32) => TestResult::TrFailed,
   |                           -^^ expected one of `)`, `,`, `...`, `..=`, `..`, or `|`
   |                           help: missing `,`


error: expected one of `)`, `,`, or `|`, found `i32`
   |
   |
96 |         Some(0xC0000409u32 as i32) => TestResult::TrFailed,
   |                              -^^^ expected one of `)`, `,`, or `|`
   |                              help: missing `,`


fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/iter/traits/mod.rs" "/checkout/library/core/src/iter/traits/collect.rs" "/checkout/library/core/src/iter/traits/unchecked_iterator.rs" "/checkout/library/core/src/iter/traits/exact_size.rs" "/checkout/library/core/src/iter/traits/iterator.rs" "/checkout/library/core/src/iter/traits/accum.rs" "/checkout/library/core/src/iter/traits/double_ended.rs" "/checkout/library/core/src/iter/range.rs" "/checkout/library/core/src/iter/sources.rs" "/checkout/library/core/src/lib.miri.rs" "/checkout/library/core/src/ffi/mod.rs" "/checkout/library/core/src/ffi/c_str.rs" "/checkout/library/core/src/ffi/va_list.rs" "/checkout/library/core/src/ascii/ascii_char.rs" "/checkout/library/core/src/bool.rs" "/checkout/library/core/src/escape.rs" "/checkout/library/core/src/arch.rs" "/checkout/library/core/src/sync/exclusive.rs" "/checkout/library/core/src/sync/mod.rs" "/checkout/library/core/src/sync/atomic.rs" "/checkout/library/rustc-std-workspace-alloc/lib.rs" "/checkout/library/core/src/fmt/num.rs" "/checkout/library/core/src/fmt/rt.rs" "/checkout/library/core/src/fmt/mod.rs" "/checkout/library/core/src/fmt/nofloat.rs" "/checkout/library/core/src/fmt/float.rs" "/checkout/library/core/src/fmt/builders.rs" "/checkout/library/core/src/mem/manually_drop.rs" "/checkout/library/core/src/mem/transmutability.rs" "/checkout/library/core/src/mem/mod.rs" "/checkout/library/core/src/mem/maybe_uninit.rs" "/checkout/library/core/src/intrinsics.rs" "/checkout/library/core/src/ub_checks.rs" "/checkout/library/test/src/cli.rs" "/checkout/library/test/src/term.rs" "/checkout/library/test/src/tests.rs" "/checkout/library/test/src/test_result.rs" "/checkout/library/core/src/array/ascii.rs" "/checkout/library/core/src/array/drain.rs" "/checkout/library/core/src/array/mod.rs" "/checkout/library/core/src/array/equality.rs" "/checkout/library/core/src/array/iter.rs" "/checkout/library/core/src/intrinsics/simd.rs" "/checkout/library/core/src/intrinsics/mir.rs" "/checkout/library/test/src/formatters/json.rs" "/checkout/library/test/src/formatters/mod.rs" "/checkout/library/test/src/formatters/junit.rs" "/checkout/library/test/src/formatters/pretty.rs" "/checkout/library/test/src/formatters/terse.rs" "/checkout/library/core/src/hash/sip.rs" "/checkout/library/core/src/hash/mod.rs" "/checkout/library/test/src/helpers/metrics.rs" "/checkout/library/test/src/helpers/mod.rs" "/checkout/library/test/src/helpers/shuffle.rs" "/checkout/library/test/src/helpers/concurrency.rs" "/checkout/library/test/src/bench.rs" "/checkout/library/test/src/lib.rs" "/checkout/library/core/src/alloc/mod.rs" "/checkout/library/core/src/alloc/global.rs" "/checkout/library/core/src/alloc/layout.rs" "/checkout/library/test/src/event.rs" "/checkout/library/test/src/types.rs" "/checkout/library/test/src/stats/tests.rs" "/checkout/library/core/src/iter/traits/marker.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Jul 11 05:35:56 UTC 2024
  network time: Thu, 11 Jul 2024 05:35:56 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/test/src/test_result.rs
+++ b/library/test/src/test_result.rs
@@ -21,6 +21,14 @@ pub const TR_OK: i32 = 50;
 #[cfg(windows)]
 const STATUS_ABORTED: i32 = 0xC0000409u32 as i32;
 
+// On Zircon (the Fuchsia kernel), an abort from userspace calls the
+// LLVM implementation of __builtin_trap(), e.g., ud2 on x86, which
+// raises a kernel exception. If a userspace process does not
+// otherwise arrange exception handling, the kernel kills the process
+// with this return code.
+#[cfg(target_os = "fuchsia")]
+const ZX_TASK_RETCODE_EXCEPTION_KILL: i32 = -1028;
+
 #[derive(Debug, Clone, PartialEq)]
 pub enum TestResult {
     TrOk,
@@ -105,6 +113,9 @@ pub fn get_result_from_exit_code(
             }
             None => unreachable!("status.code() returned None but status.signal() was None"),
         },
+        // Upon an abort, Fuchsia returns the status code ZX_TASK_RETCODE_EXCEPTION_KILL.
+        #[cfg(target_os = "fuchsia")]
+        Some(ZX_TASK_RETCODE_EXCEPTION_KILL) => TestResult::TrFailed,
         #[cfg(not(unix))]
         None => TestResult::TrFailedMsg(format!("unknown return code")),
         #[cfg(any(windows, unix))]

--- a/library/test/src/test_result.rs
+++ b/library/test/src/test_result.rs
@@ -19,7 +19,7 @@ pub const TR_OK: i32 = 50;
 // On Windows we use __fastfail to abort, which is documented to use this
 // exception code.
 #[cfg(windows)]
-const STATUS_ABORTED: i32 = 0xC0000409u32 as i32;
+const STATUS_FAIL_FAST_EXCEPTION: i32 = 0xC0000409u32 as i32;
 
 // On Zircon (the Fuchsia kernel), an abort from userspace calls the
 // LLVM implementation of __builtin_trap(), e.g., ud2 on x86, which
@@ -104,7 +104,7 @@ pub fn get_result_from_exit_code(
     let result = match status.code() {
         Some(TR_OK) => TestResult::TrOk,
         #[cfg(windows)]
-        Some(STATUS_ABORTED) => TestResult::TrFailed,
+        Some(STATUS_FAIL_FAST_EXCEPTION) => TestResult::TrFailed,
         #[cfg(unix)]
         None => match status.signal() {
             Some(libc::SIGABRT) => TestResult::TrFailed,

--- a/tests/ui/test-attrs/test-panic-abort-nocapture.rs
+++ b/tests/ui/test-attrs/test-panic-abort-nocapture.rs
@@ -10,7 +10,6 @@
 //@ ignore-wasm no panic or subprocess support
 //@ ignore-emscripten no panic or subprocess support
 //@ ignore-sgx no subprocess support
-//@ ignore-fuchsia code returned as ZX_TASK_RETCODE_EXCEPTION_KILL, FIXME (#127539)
 
 #![cfg(test)]
 

--- a/tests/ui/test-attrs/test-panic-abort-nocapture.run.stderr
+++ b/tests/ui/test-attrs/test-panic-abort-nocapture.run.stderr
@@ -1,9 +1,9 @@
-thread 'main' panicked at $DIR/test-panic-abort-nocapture.rs:35:5:
+thread 'main' panicked at $DIR/test-panic-abort-nocapture.rs:34:5:
 assertion `left == right` failed
   left: 2
  right: 4
 note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
-thread 'main' panicked at $DIR/test-panic-abort-nocapture.rs:29:5:
+thread 'main' panicked at $DIR/test-panic-abort-nocapture.rs:28:5:
 assertion `left == right` failed
   left: 2
  right: 4

--- a/tests/ui/test-attrs/test-panic-abort.rs
+++ b/tests/ui/test-attrs/test-panic-abort.rs
@@ -10,7 +10,6 @@
 //@ ignore-wasm no panic or subprocess support
 //@ ignore-emscripten no panic or subprocess support
 //@ ignore-sgx no subprocess support
-//@ ignore-fuchsia code returned as ZX_TASK_RETCODE_EXCEPTION_KILL, FIXME (#127539)
 
 #![cfg(test)]
 #![feature(test)]

--- a/tests/ui/test-attrs/test-panic-abort.run.stdout
+++ b/tests/ui/test-attrs/test-panic-abort.run.stdout
@@ -17,7 +17,7 @@ hello, world
 testing123
 ---- it_fails stderr ----
 testing321
-thread 'main' panicked at $DIR/test-panic-abort.rs:40:5:
+thread 'main' panicked at $DIR/test-panic-abort.rs:39:5:
 assertion `left == right` failed
   left: 2
  right: 5

